modules/phase_field/examples/rigidbodymotion/AC_CH_Multigrain.i

```diff
@@ -0,0 +1,299 @@
+  [../]
+[]
+
+[AuxVariables]
+  [./bnds]
+  [../]
+  [./MultiAuxVariables]
+    order = CONSTANT
+    family = MONOMIAL
+    var_name_base = 'dfx dfy' # Names of force density variables
+    op_num = '4 4' # Should both be equal to global op_num
+  [../]
+  [./force]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./free_energy]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+[]
+
+[Functions]
+  [./load_x]
+    # Defines the force on the grains in the x-direction
+    type = ParsedFunction
+    value = 0.005*cos(x*pi/600)
+  [../]
+  [./load_y]
+    # Defines the force on the grains in the y-direction
+    type = ConstantFunction
+    value = 0.002
+  [../]
+[]
+
+[Kernels]
+  [./RigidBodyMultiKernel]
+    # Creates all of the necessary Allen Cahn kernels automatically
+    c = c
+    f_name = f_loc
+    mob_name = L
+    kappa_name = kappa_gr
+  [../]
+  # Cahn Hilliard kernels
+  [./dt_w]
+    type = CoupledTimeDerivative
+    variable = w
+    v = c
+  [../]
+  [./CH_wres]
+    type = SplitCHWRes
+    variable = w
+    mob_name = M
+  [../]
+  [./CH_Parsed]
+    type = SplitCHParsed
+    variable = c
+    f_name = f_loc
+    w = w
+    kappa_name = kappa_c
+    args = 'gr0 gr1 gr2 gr3' # Must be changed as op_num changes. Copy/paste from line 4
+  [../]
+  [./CH_RBM]
+    type = MultiGrainRigidBodyMotion
+    variable = w
+    c = c
+    v = 'gr0 gr1 gr2 gr3' # Must be changed as op_num changes. Copy/Paste from line 4
+  [../]
+[]
+
+[AuxKernels]
+  [./force_x]
+    type = FunctionAux
+    variable = force
+    function = load_x
+  [../]
+  [./force_y]
+    type = FunctionAux
+    variable = force
+    function = load_y
+  [../]
+  [./MatVecRealGradAuxKernel]
+    var_name_base = 'dfx dfy'    # Names of force density variables
+    op_num = '4 4'               # Should both be equal to global op_num
+    property = force_density_ext
+  [../]
+  [./energy_density]
+    type = TotalFreeEnergy
+    variable = free_energy
+    f_name = f_loc
+    kappa_names = kappa_c
+    interfacial_vars = c
+  [../]
+  [./bnds]
+    type = BndsCalcAux
+    variable = bnds
+    v = 'gr0 gr1 gr2 gr3' # Must be changed as op_num changes. Copy/Paste from line 4
+  [../]
+[]
+
+[BCs]
+  [./bcs]
+    #zero flux BC
+    type = NeumannBC
+    value = 0
+    variable = c
+    boundary = '0 1 2 3'
+  [../]
+[]
+
+[Materials]
+  [./constants]
+    type = GenericConstantMaterial
+    block = 0
+    prop_names = 'kappa_gr kappa_c M L'
+    prop_values = '50 3000 4.5 60'
+  [../]
+  [./free_energy]
+    type = DerivativeParsedMaterial
+    block = 0
+    f_name = f_loc
+    constant_names = 'A B'
+    constant_expressions = '450 1.5'
+    args = 'c gr0 gr1 gr2 gr3' #Must be changed as op_num changes. Copy/paste from line 4
+    function = 'A*c^2*(1-c)^2+B*(c^2+6*(1-c)*(gr0^2+gr1^2+gr2^2+gr3^2)
+                -4*(2-c)*(gr0^3+gr1^3+gr2^3+gr3^3)
+                +3*(gr0^2+gr1^2+gr2^2+gr3^2)^2)'
+                                 #Copy/paste from lines 5-6
```

Comment:
One issue with generating the expression with the action is that the first part of the function 
`A*c^2*(1-c)^2` 
is a free-standing free energy function and should be changed by the user to represent a real material. I guess it would be possible to make that term an input and the rest of it generate automatically. I just don't know if I can code in MOOSE well enough to make that work.